corehq/apps/case_search/utils.py

```diff
@@ -261,34 +275,71 @@ def get_related_cases(helper, app_id, case_types, cases, custom_related_case_pro
         return []
 
     app = get_app_cached(helper.domain, app_id)
-    paths = [
-        rel for rels in [get_related_case_relationships(app, case_type) for case_type in case_types]
-        for rel in rels
-    ]
-    child_case_types = [
-        _type for types in [get_child_case_types(app, case_type) for case_type in case_types]
-        for _type in types
-    ]
 
     expanded_case_results = []
     if custom_related_case_property:
         expanded_case_results.extend(get_expanded_case_results(helper, custom_related_case_property, cases))
 
     results = expanded_case_results
     top_level_cases = cases + expanded_case_results
-    if paths:
-        results.extend(get_related_case_results(helper, top_level_cases, paths))
 
-    if child_case_types:
-        results.extend(get_child_case_results(helper, top_level_cases, child_case_types))
+    defined_cases = get_related_cases_result(helper, app, case_types, top_level_cases, include_related_cases)
```

Comment:
nit: rename `defined_cases` to `related_cases`

Reply:
renamed [ccc06cf](https://github.com/dimagi/commcare-hq/pull/32587/commits/ccc06cf9364a69edfb8ffa924042c56afb4baa3e)